libs/dal/src/dal.service.ts

```diff
@@ -1,16 +1,17 @@
 import mongoose, { Connection, ConnectOptions } from 'mongoose';
 
+export const baseConfig: ConnectOptions = {
+  // AUTO_CREATE_INDEXES is deprecated, use MONGO_AUTO_CREATE_INDEXES
+  autoIndex: process.env.AUTO_CREATE_INDEXES === 'true' || process.env.MONGO_AUTO_CREATE_INDEXES === 'true',
+  maxIdleTimeMS: process.env.MONGO_MAX_IDLE_TIME_IN_MS ? Number(process.env.MONGO_MAX_IDLE_TIME_IN_MS) : 1000 * 60,
+  maxPoolSize: process.env.MONGO_MAX_POOL_SIZE ? Number(process.env.MONGO_MAX_POOL_SIZE) : 50,
+  minPoolSize: process.env.MONGO_MIN_POOL_SIZE ? Number(process.env.MONGO_MIN_POOL_SIZE) : 10,
```

Comment:
We need to the `MONGO_AUTO_CREATE_INDEXES` and `MONGO_MAX_IDLE_TIME_IN_MS` env vars to the `env.validators.ts` file in API ([example for API](https://github.com/novuhq/novu/blob/next/apps/api/src/config/env.validators.ts)), Worker, Websocket, and Webhook to ensure they are valid values during application startup.

Then we can populate all the defaults in there, avoiding env var defaults in the code.

Reply:
Good point! :)

Reply:
#til about these validators. Excellent callout! Fixed.